Build fails with arrow@4.3

**Describe the bug**
[arrow v4.3](https://github.com/apache/arrow-datafusion/blob/master/datafusion/Cargo.toml#L49) is currently specified as a dependency, but it looks like v4.4 is required:

```
   Compiling datafusion v4.0.0-SNAPSHOT (/home/todd/arrow-datafusion/datafusion)
error[E0308]: mismatched types
   --> datafusion/src/physical_plan/expressions/nth_value.rs:189:27
    |
189 |                     shift(arr.as_ref(), index as i64).map_err(DataFusionError::ArrowError)
    |                           ^^^^^^^^^^^^ expected struct `PrimitiveArray`, found trait object `dyn arrow::array::Array`
    |
    = note: expected reference `&PrimitiveArray<_>`
               found reference `&dyn arrow::array::Array`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0308`.
error: could not compile `datafusion`

To learn more, run the command again with --verbose.
```


**To Reproduce**
Steps to reproduce the behavior:
To reproduce you might need to force the arrow and parquet dependencies to `4.3.0`, and then run `cargo build`:

```
arrow = { version = "=4.3", features = ["prettyprint"] }
parquet = { version = "=4.3", features = ["arrow"] }
```


**Expected behavior**
For the build to succeed with the specified dependencies.

--- a/datafusion/Cargo.toml
+++ b/datafusion/Cargo.toml
@@ -46,8 +46,8 @@ unicode_expressions = ["unicode-segmentation"]
 [dependencies]
 ahash = "0.7"
 hashbrown = "0.11"
-arrow = { version = "4.3", features = ["prettyprint"] }
-parquet = { version = "4.3", features = ["arrow"] }
+arrow = { version = "4.4", features = ["prettyprint"] }
+parquet = { version = "4.4", features = ["arrow"] }
 sqlparser = "0.9.0"
 paste = "^1.0"
 num_cpus = "1.13.0"

